app/tests/components_tests/test_tasks.py

```diff
@@ -463,35 +470,30 @@ def test_add_image_to_object_marks_job_as_failed_on_validation_fail(
     settings.task_eager_propagates = (True,)
     settings.task_always_eager = (True,)
 
-    obj = AlgorithmJobFactory(time_limit=10)
+    job = AlgorithmJobFactory(time_limit=10)
     us = RawImageUploadSessionFactory(status=RawImageUploadSession.SUCCESS)
     ci = ComponentInterfaceFactory(kind=InterfaceKindChoices.PANIMG_IMAGE)
 
-    error_message = f"Image validation for socket {ci.title} failed with error: Image imports should result in a single image. "
-
     linked_task = some_async_task.signature(
         kwargs={"foo": "bar"}, immutable=True
     )
 
     with django_capture_on_commit_callbacks(execute=True) as callbacks:
         add_image_to_object(
-            app_label=obj._meta.app_label,
-            model_name=obj._meta.model_name,
+            app_label=job._meta.app_label,
+            model_name=job._meta.model_name,
             upload_session_pk=us.pk,
-            object_pk=obj.pk,
+            object_pk=job.pk,
             interface_pk=ci.pk,
             linked_task=linked_task,
         )
 
     assert ComponentInterfaceValue.objects.filter(interface=ci).count() == 0
-    us.refresh_from_db()
-    assert us.status == RawImageUploadSession.FAILURE
-    assert us.error_message == error_message
-    obj.refresh_from_db()
-    assert obj.status == obj.CANCELLED
-    assert obj.error_message == "One or more of the inputs failed validation."
+    job.refresh_from_db()
+    assert job.status == job.CANCELLED
+    assert job.error_message == "One or more of the inputs failed validation."
```

Comment:
Why has the test on the upload session been removed? The upload session should still be marked as failed. A job can have multiple image inputs, and thus multiple upload sessions associated with it. The user would need to know which one of those upload sessions failed.